sqlalchemy-stubs/sql/sqltypes.pyi

```diff
@@ -183,7 +187,7 @@ class SchemaType(SchemaEventTarget):
         _create_events: bool = ...,
     ) -> None: ...
     def copy(self, **kw: Any): ...
-    def adapt(self, impltype: Any, **kw: Any): ...
+    def adapt(self, __impltype: Type[_T], **kw: Any) -> _T: ...
```

Comment:
what's the rationale to rename this to `__impltype` ?

Reply:
For function signatures in stub files, variables prefixed with `__` are treated as positional-only. Because `TypeEngine` uses `cls` for this parameter and `SchemaType` (and others) use `impltype`, `mypy` marks them as incompatible because technically you could call the method using `adapt(cls=Thing)` or `adapt(impltype=Thing)`. However, it's pretty clear from how `adapt()` is used in the codebase that it should be positional-only and then `mypy` ignores the name of the parameter when comparing them.

Reply:
oh nice!  yes, for SQLAlchemy 2.0 i am SUPER looking forward to positional-only and keyword-only.      although folks might not like me changing some of those things.

Reply:
I've got some personal projects I've migrated to `3.9` and the two things I love are positional-only params and union types using `|`

Reply:
I guess also base type being generic would be nice, like `list[int]`

Reply:
> I guess also base type being generic would be nice, like `list[int]`

There are still some issues using subscriptable builtins in stubs, but there are places where we could definitely do that

Reply:
I meant things that are nice in py3.9+, not necessary to use in the this typing